pom.xml

```diff
@@ -1770,6 +1770,10 @@
             <groupId>org.apache.hive</groupId>
             <artifactId>hive-storage-api</artifactId>
           </exclusion>
+          <exclusion>
+            <groupId> com.esotericsoftware</groupId>
+            <artifactId>kryo-shaded</artifactId>
```

Comment:
why this change?

Reply:
CC @wangyum yes good point. ORC also uses kryo-shaded, and uses 3.0.3. In theory that could cause a break, but the tests pass. That's a positive sign but not bulletproof. @dongjoon-hyun do you have any insight into how ORC uses kryo? Is it a code path that wouldn't matter to Spark?

Reply:
Thank you for pinging me, @srowen . ORC uses Kryo only for writing/reading one ORC configuration, `orc.kryo.sarg`. The followings are the Spark's code indirect code path. I guess Kryo provides backward compatibility at lest, but I'll take a look at this PR today.
- For new ORCFileFormat, [OrcInputFormat.setSearchArgument](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala#L159)
- For old ORCFileFormat, [SearchArgument.toKryo](https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala#L129)

Reply:
@srowen  In short, the current Spark always uses the same Kryo version for read/write `SearchArgument` and it's used only on runtime.

1. Old OrcFileFormat always uses `org.spark-project.hive:hive-exec:1.2.1.spark2` which uses the shaded one in `hive-exec`.
    - `com.esotericsoftware.kryo:kryo:2.21`. 

2. New OrcFileFormat uses `org.apache.orc` which uses the one provided by Spark.
    - `com.esotericsoftware:kryo-shaded:3.0.3`  (All Spark/Orc/Hive uses this version for now)

3. New OrcFileFormat (in this PR) uses `org.apache.orc` which uses the one provided by Spark.
    - `com.esotericsoftware:kryo-shaded:4.0.2`

So, (1) is unchanged by this PR. (2) and (3) also doesn't use a mixed version of Kryo. So, it should be fine because Apache Spark doesn't allow a mixed Spark version(master and executor). BTW, during investigation, there was some performance issue in `createFilter`. I'll file a new JIRA for that.

Reply:
I guess my question is whether Orc complied against kryo-shaded 3.x necessarily works with kryo-shaded 4.x, because there were API and behavior changes. Our current tests don't seem to surface any such problems, but who knows if there's something they don't test.

I agree that our distribution won't include two versions, but the issue I'm wondering about is whether Orc will like the one later version that it gets bundled with.

Reply:
Yes. I checked that, @srowen . `org.apache.orc` only uses Kryo constructor, `writeObject`, and `readObject` from `kryo-shaded` library. There is no change for them.

**WRITE**
```
(new Kryo()).writeObject(out, sarg);
```
**READ**
```
... = (new Kryo()).readObject(new Input(sargBytes), SearchArgumentImpl.class);
```

Reply:
OK, if that's the extent of the usage, then I believe that ORC is OK with Kryo 4. That much is not a problem.

Reply:
I was also worried that part, but it's used only in a run-time `SearchArgument` serialization. There was no usage with ORC `file`s.